script.py

```diff
@@ -0,0 +1,98 @@
+import json
+import requests
+
+import config
+
+
+def main():
+    if token:
+        response = request_vehicles(token)
+        print_info(response)
+        return 0
+
+    else:
+        return 1
+
+
+def generate_token(client_id, client_secret):
+    token_response = requests.post(f'{osu_API}oauth2/token', data={
+        'client_id': client_id,
+        'client_secret': client_secret,
+        'grant_type': 'client_credentials'
+    })
+
+    if(token_response.status_code == 200):
+        return token_response.json()['access_token']
+    else:
+        print('Invalid credentials')
+
+
+def request_vehicles(token):
+    return requests.get(f'{osu_API}v1/beaverbus/vehicles', headers={
+        'Authorization': f'Bearer {token}'
+    })
+
+
+def request_vehicle_by_ID(token, ID):
+    return requests.get(f'{osu_API}v1/beaverbus/vehicles/{ID}', headers={
+        'Authorization': f'Bearer {token}'
+    })
+
+
+def print_info(response):
+    response_data_JSON = response.json()['data']
```

Comment:
What's the purpose of querying it before receiving an input from the client? I guess it kind of makes sense to print all of the running vehicles IDs and let your users know what are available buses to query.

In addition, when I was testing this script, there's no beaver bus running (actually it's a GPS issue:

> about 3 hours ago: The Central route is currently running, the bus is having intermittent gps issues.

) and I got the following response with the status code 200 when requesting `/beaverbus/vehicles`:

```json
{
    "links": {},
    "data": []
}
```

But the script didn't tell me there is no bus available to search, instead, it still prompts me to enter input. I think it'll make more sense to print some messages if the data array is empty.